bf2fc6cc711aee1a0c2a/kas-fleet-manager
# feat: redefine enterprise cluster endpoints response behavior

## Description

Related to https://issues.redhat.com/browse/MGDSTRM-10809.

After some discussions with team members around the data returned in the enterprise cluster related endpoints
this PR changes the information they return in the following way:

* The supported instance types and the capacity information is now returned in the Register, Get and
  List Enterprise Cluster endpoints
* Additionally, in the Register enterprise cluster the fleetshard parameters are returned
* The Enterprise Cluster addon parameters endpoint has been changed to now:
  * Return Kind: ClusterAddonParameters
  * Return ID: <cluster_id> of the corresponding Enterprise Clutser
  * Return Href: The URL of the addon parameters endpoint called
  * Return The list of fleetshard parameters
  * Other information about the cluster is not returned anymore

The reason this changes have been made is due to before the set of information returned between the different enterprise cluster endpoints was inconsistent. The reason for that was due to having a set of requirements that have changed over time. This PR makes the behavior between the endpoints consistent and adapted to the current requirements/needs.

## Verification Steps
* Verify enterprise cluster registration endpoint returns the enterprise cluster information, including the fleetshard parameters. Additionally, now it should also include the supported instance types and the capacity information
* Verify both the enterprise cluster get and list endpoints return the enterprise cluster information, now also including the supported instance types and the capacity information. The fleetshard parameters should not be shown as part of the results
* Verify that the enterprise cluster addon parameters endpoint now returns the following
  * kind: "ClusterAddonParameters"
  * id: <cluster_id>
  * href: the url of the addon parameters endpoint called
  * fleetshard_parameters: The list of fleetshard parameters
  * Other information about the cluster itself is not returned anymore


## Checklist (Definition of Done)
